src/main/java/com/example/Library/services/AuthenticationService.java

```diff
@@ -0,0 +1,51 @@
+package com.example.Library.services;
+
+import com.example.Library.dto.AuthenticationRequestDTO;
+import com.example.Library.model.User;
+import com.example.Library.repositories.UserRepository;
+import com.example.Library.security.JwtTokenProvider;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Service
+public class AuthenticationService {
+
+    public AuthenticationService(AuthenticationManager authenticationManager, UserRepository userRepository, JwtTokenProvider jwtTokenProvider) {
+        this.authenticationManager = authenticationManager;
+        this.userRepository = userRepository;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    private final AuthenticationManager authenticationManager;
+    private UserRepository userRepository;
+    private JwtTokenProvider jwtTokenProvider;
+
+    public ResponseEntity<?> authenticate(@RequestBody AuthenticationRequestDTO request) {
```

Comment:
1) It may be better to introduce new DTO named AuthenticationResponseDTO (with email & token fields)
2) It would be better to simply return AuthenticationResponseDTO here (Authentication service should not return ResponseEntity because it is a presentation-layer class)